### PR TITLE
Update to track scan limit in android

### PR DESCRIPTION
**Pull request by brookflok**
Created a simple boolean function that tracks the error for too many scan attempts and handles that as a promise.

### CODE REVIEW

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java [hunk 1]
```diff
@@ -397,9 +399,16 @@ public void failed(@NonNull RemoveErrorCode errorCode) {
...
     public void reScanAndLoadWifiList(final Promise promise) {
+        boolean wifiStartScan = wifi.startScan();
+        Log.d(TAG, "wifi start scan: " + wifiStartScan);
+        if(wifiStartScan == true){
         final WifiScanResultReceiver wifiScanResultReceiver = new WifiScanResultReceiver(wifi, promise);
         getReactApplicationContext().registerReceiver(wifiScanResultReceiver, new IntentFilter(WifiManager.SCAN_RESULTS_AVAILABLE_ACTION));
-        wifi.startScan();
+        }else{
+            Log.d(TAG, "Wifi scan rejected");
+            promise.resolve("Starting Android 9, it's only allowed to scan 4 times per 2 minuts in a foreground app.");
```
Comment (hakansaglam29) [inline android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java:409]: "Starting Android 9, it's only allowed to scan 4 times per 2 minutes in a foreground app."

**Comment by eliaslecomte** [comment]
Actually the implementation has been changed with the 4.7.0 release to no longer scan while connecting. So that makes this mr unneeded?

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java [hunk 2]
```diff
@@ -393,9 +395,16 @@ public void failed(@NonNull RemoveErrorCode errorCode) {
      */
     @ReactMethod
     public void reScanAndLoadWifiList(final Promise promise) {
+        boolean wifiStartScan = wifi.startScan();
```
Comment (eliaslecomte) [inline android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java:398]: wow I didn't know you could ask this upfront :-O

**Comment by brookflok** [comment]
> Thanks for your contribution 🎉🥳, The actual implementation has been changed with the 4.7.0 release to no longer scan while connecting. So that makes this pr unneeded?

Well, I think it is good to give a user response as a promise for too many scan attempts. I m using my code throughout the app and it is helping me to improve my error cases when a user scans to many times.

**Review by eliaslecomte — APPROVED** [review]

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java [hunk 3]
```diff
@@ -381,13 +383,19 @@ public void failed(@NonNull RemoveErrorCode errorCode) {
...
-        getReactApplicationContext().registerReceiver(wifiScanResultReceiver, new IntentFilter(WifiManager.SCAN_RESULTS_AVAILABLE_ACTION));
-        wifi.startScan();
+        boolean wifiStartScan = wifi.startScan();
+        Log.d(TAG, "wifi start scan: " + wifiStartScan);
+        if (wifiStartScan == true) {
+          final WifiScanResultReceiver wifiScanResultReceiver = new WifiScanResultReceiver(wifi, promise);
+          getReactApplicationContext().registerReceiver(wifiScanResultReceiver, new IntentFilter(WifiManager.SCAN_RESULTS_AVAILABLE_ACTION));
+        } else {
+            Log.d(TAG, "Wifi scan rejected");
+            promise.resolve("Starting Android 9, it's only allowed to scan 4 times per 2 minuts in a foreground app.");
```
Comment (robrechtme) [inline android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java:397]: Makes more sense to reject the promise?

**Comment by JuanSeBestia** [comment]
@brookflok do you want be a maintainer of this repo also ?

**Comment by brookflok** [comment]
> @brookflok do you want be a maintainer of this repo also ?

Yea, why not? I would love to follow this

**Pull request merged**